canonical/maas-ui
# feat: esc button to close side panel MAASENG-1590

## Done

- feat: esc button to close side panel MAASENG-1590
- fix: ignore the "[" keyboard shortcut when modifier key is pressed
- fix: add missing side panel header titles
- refactor to use a single SidePanelContext in preparation for restructuring the component tree in https://warthogs.atlassian.net/browse/MAASENG-1245

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch


You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open side panel on every page and verify you can close it by pressing the "Esc" button
- Verify that once opened, a side panel gets closed automatically when you navigate to a different page
- press "[" on your keyboard and verify the side navigation has been toggled
- press "ctrl+[" on your keyboard and verify that the side navigation state has not changed


## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-1590

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots
### Network discovery side panel
#### Before
<img width="1278" alt="Google Chrome screenshot 000143@2x" src="https://github.com/canonical/maas-ui/assets/7452681/f1938589-bf29-4536-9bc5-32215ec271d2">

#### After
<img width="1277" alt="Google Chrome screenshot 000141@2x" src="https://github.com/canonical/maas-ui/assets/7452681/f5b430e4-b7bc-41b6-a0a6-e2061429f950">

### Delete VLAN side panel
#### Before
<img width="1452" alt="Google Chrome screenshot 000147@2x" src="https://github.com/canonical/maas-ui/assets/7452681/8a2d5e81-7358-4c7a-a9c7-4ce7b5de9899">

#### After
<img width="1450" alt="Google Chrome screenshot 000145@2x" src="https://github.com/canonical/maas-ui/assets/7452681/b7a61bc2-a776-4180-ae0f-1b0383231dc3">
